### PR TITLE
Fix URLs in appeals API docs

### PR DESCRIPTION
**Pull request by caseywilliams**
## Summary

- In some cases here, we did not initially know what the final URLs would be
- In others, the format of the URLs has changed since we first put these markdown files together
- I also made a few fixes to mentions of evidence submission routes

## Related issue(s)
- [Slack thread](https://lighthouseva.slack.com/archives/GPX76BVKN/p1692718758017309)

## Testing done

- Clicked on all dev URLs in this PR manually to load them in a browser
- Visually confirmed the dev URLs are equivalent to the prod ones (there are not live yet so I can't check them manually)

## What areas of the site does it impact?
- v0 appeals APIs

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

File: modules/appeals_api/app/swagger/legacy_appeals/v0/api_description.md [hunk 1]
```diff
@@ -1,18 +1,18 @@
-The Legacy Appeals API returns a list of a claimant's active legacy appeals, which are not part of the Appeals Modernization Act (AMA) process. This list can be used to determine whether to opt in to the new decision review process. [Learn more about managing a legacy appeal.](https://www.va.gov/decision-reviews/legacy-appeals/).
+The Legacy Appeals API returns a list of a claimant's active legacy appeals, which are not part of the Appeals Modernization Act (AMA) process. This list can be used to determine whether to opt in to the new decision review process. [Learn more about managing a legacy appeal](https://www.va.gov/decision-reviews/legacy-appeals/).
```
Comment (kristen-brown) [inline modules/appeals_api/app/swagger/legacy_appeals/v0/api_description.md:1]: This change moved the `.`, which I think should be part of the link (as it was before).

File: modules/appeals_api/app/swagger/legacy_appeals/v0/api_description_dev.md [hunk 2]
```diff
@@ -1,18 +1,18 @@
-The Legacy Appeals API returns a list of a claimant's active legacy appeals, which are not part of the Appeals Modernization Act (AMA) process. This list can be used to determine whether to opt in to the new decision review process. [Learn more about managing a legacy appeal.](https://www.va.gov/decision-reviews/legacy-appeals/).
+The Legacy Appeals API returns a list of a claimant's active legacy appeals, which are not part of the Appeals Modernization Act (AMA) process. This list can be used to determine whether to opt in to the new decision review process. [Learn more about managing a legacy appeal](https://www.va.gov/decision-reviews/legacy-appeals/).
```
Comment (kristen-brown) [inline modules/appeals_api/app/swagger/legacy_appeals/v0/api_description_dev.md:1]: Same comment as above.

File: modules/appeals_api/app/swagger/notice_of_disagreements/v0/api_description.md [hunk 3]
```diff
@@ -36,7 +36,7 @@ Statuses can also be simulated for evidence document uploads.
 
 ### Evidence uploads
 Our NOD evidence submission endpoints allow a client to upload a document package (documents and metadata) of supporting evidence for their submission by following these steps.
-1. Use the endpoint `POST /evidence_submissions` to return a JSON service response with the attributes listed below.
+1. Use the endpoint `POST /evidence-submissions` to return a JSON service response with the attributes listed below.
```
Comment (kristen-brown) [inline modules/appeals_api/app/swagger/notice_of_disagreements/v0/api_description.md:39]: Good catch.

**Review by kristen-brown — COMMENTED** [review]
Looks good! I double-checked all the links and also compared them to their file names (e.g. `higher-level-reviews` links in the `higher_level_reviews` descriptions).

I left a couple of comments about the `.` being in the wrong place for links that are an entire sentence.

File: modules/appeals_api/app/swagger/legacy_appeals/v0/api_description.md (hunk 1 above)
Comment (caseywilliams) [reply modules/appeals_api/app/swagger/legacy_appeals/v0/api_description.md:1]: I did this on purpose - I don't think periods are supposed to be included in links in cases where the link text doesn't include the entire sentence - here are two places I found that mention it:
* https://ux.stackexchange.com/questions/17331/should-a-full-sentence-html-link-include-the-period-in-the-linked-text
* https://www.chicagomanualofstyle.org/qanda/data/faq/topics/URLs/faq0007.html

That said, google's style guide suggests avoiding this confusion by moving the link out of the main text entirely: https://developers.google.com/style/periods#periods-with-urls - it seems like we'd need to go through UX to take that route, though

**Review by kristen-brown — APPROVED** [review]

**Pull request merged**